### PR TITLE
rebase changes with PR:707

### PR DESCRIPTION
**Pull request by tpereirasplunk**
Fix Issues with pipeline

Int test run: https://github.com/splunk/splunk-operator/runs/6011142014?check_suite_focus=true

### CODE REVIEW

**Review by jambrosiano — APPROVED** [review]

**Review by pdhanoyasplunk — APPROVED** [review]

File: test/testenv/deployment.go [hunk 1]
```diff
@@ -759,8 +759,6 @@ func (d *Deployment) DeployClusterMasterWithGivenSpec(ctx context.Context, name
 	if err != nil {
 		return nil, err
 	}
-	// Verify standalone goes to ready state
-	ClusterManagerReady(ctx, d, d.testenv)
```
Comment (smohan-splunk) [inline test/testenv/deployment.go:763]: Can you please clarify why this change?

Comment (tpereirasplunk) [reply test/testenv/deployment.go:763]: This was added as @vivekr-splunk indicated it is highly unlikely for customers to deploy both CM and SHC at the same time as we do in automation. They would wait for CM to be ready before doing the same.

However if we wait for CM to be ready most times the app are downloaded and about to be installed. We have tests that require us to interrupt the operator in the download phase which would be missed and the test would fail.

Comment (sgontla) [reply test/testenv/deployment.go:763]: @tpereirasplunk, the App framework should not come into the timelines and the way the SHC connects CM, as the Phase-3 app framework always waits for the corresponding Pod to come up, before starting the app framework. In phase-2 it may be influencing the sequence. I wonder if @vivekr-splunk  did this change considering the phase-2 app framework which was still the case with the 1.1.0 release.

Comment (tpereirasplunk) [reply test/testenv/deployment.go:763]: @sgontla . The issue with this change was that we waited for CM to be ready before deploying SHC. So by the time we have CM ready and SHC deployed, the phase on CM is post download phase.

And we needed to interrupt the operator in download phase for one of the test case. So we usually call the function to check if CM is ready after in a separate step

Comment (tpereirasplunk) [reply test/testenv/deployment.go:763]: This call is already happening later after all specs are deployed and would be a duplicate if made here

**Review by sgontla — APPROVED** [review]

**Review by smohan-splunk — APPROVED** [review]
